src/api/app/controllers/application_controller.rb

```diff
@@ -569,6 +569,17 @@ def set_response_format_to_xml
     request.format = :xml if request.format == :html
   end
 
+  def extract_user_public
```

Comment:
Isn't better if this method lives only in the Project::PublicKeyController ??, is only used there. 
Or you plan to use it in another controller?.

BTW, I know that you only moved the code, but... can be better like that?
```ruby
def extract_user_public
  if ::Configuration.anonymous
    load_nobody
  else
    logger.error 'No public access is configured'
    render_error( message: 'No public access is configured', status: 401 )
  end
end
```